types/tools/wrappers/base-tool-wrapper.d.ts

```diff
@@ -0,0 +1,76 @@
+import { Tool, ToolConstructable, ToolSettings } from '..';
+import type { SanitizerConfig, API as ApiMethods } from '../..';
+
+import { ToolType } from './tool-type';
+import { InlineToolWrapper } from './inline-tool-wrapper';
+import { BlockToolWrapper } from './block-tool-wrapper';
+import { BlockTuneWrapper } from './block-tune-wrapper';
+
+export interface BaseToolWrapper<Type extends ToolType, ToolClass extends Tool> {
```

Comment:
Can you describe what this interface represents and why do we need to wrap tools

Reply:
Also let's maybe call it a factory or builder, the wrapper is not really clear

Reply:
it is the question for you. It is just the type describing your BlockTool etc classes (which are actually wrappers around tools)